src/main/java/menelaus/model/board/Piece.java

```diff
@@ -63,13 +65,34 @@ public void addTile(Tile tile) {
 		tiles.add(tile);
 	}
 	
-	public Piece(Point position) {
+	public Piece(Point position, PieceDrawer pieceDrawer) {
```

Comment:
Here you are coupling your model with your drawing logic which isn't usually a good thing. Moreover, the PieceDrawer, if you were to use something like that, would probably be a view component that has access to a canvas and it would ask for multiple pieces to draw on this canvas.